archaius2-core/src/main/java/com/netflix/archaius/config/DefaultSettableConfig.java

```diff
@@ -36,16 +36,16 @@ public DefaultSettableConfig(String name) {
     public DefaultSettableConfig() {
         super(generateUniqueName("settable-"));
     }
-    
+
     @Override
     public synchronized <T> void setProperty(String propName, T propValue) {
         Map<String, Object> copy = new HashMap<>(props.size() + 1);
         copy.putAll(props);
         copy.put(propName, propValue);
-        props = copy;
+        props = Collections.unmodifiableMap(copy);
```

Comment:
Please update all the other methods that alter the internal state to do this, too.

Reply:
Good catch, I had missed that. I've updated the PR.